### ISSUE TITLE
[bug] "Add to play queue" results in queued playlist with unexpected track order when using "Play On" streaming

### ISSUE THREAD

**Issue by AlbertMarkovski** [jellyfin/jellyfin#12377]
### This issue respects the following points:

- [X] This is a **bug**, not a question or a configuration issue; Please visit our forum or chat rooms first to troubleshoot with volunteers, before creating a report. The links can be found [here](https://jellyfin.org/contact/).
- [X] This issue is **not** already reported on [GitHub](https://github.com/jellyfin/jellyfin/issues?q=is%3Aopen+is%3Aissue) _(I've searched it)_.
- [X] I'm using an up to date version of Jellyfin Server stable, unstable or master; We generally do not support previous older versions. If possible, please update to the latest version before opening an issue.
- [X] I agree to follow Jellyfin's [Code of Conduct](https://jellyfin.org/docs/general/community-standards.html#code-of-conduct).
- [X] This report addresses only a single issue; If you encounter multiple issues, kindly create separate reports for each one.

### Description of the bug

This bug cropped up for me only after completing a recent debian bullseye-to-bookworm upgrade (which also upgraded Jellyfin from 10.8.x to 10.9.8). 

This bug is NOT reproducible when queuing tracks in a browser for local audio playback. Instead, the bug only occurs when using the "Play On" streaming feature. In this example it's streaming to a Raspberry Pi running Mopidy v3.4.2 with the Mopidy-Jellyfin v1.0.5 extension. 

### Reproduction steps

1. In the upper right corner of the Jellyfin web UI, select "Play On" -> [your mopidy streaming device]
2. Find the desired first album to play in a music collection, and click play (the album is correctly queued on the remote mopidy device with album tracks in expected order, and the first track of the album correctly begins playing)
3. While the first track of the first album is playing, find the desired second album to queue following the currently playing first album, and click "Add to play queue"

### What is the current _bug_ behavior?

The resulting playlist order on the mopidy device is: 

1. myFirstAlbum - trackOne (currently playing)
2. mySecondAlbum - trackOne
3. mySecondAlbum - trackTwo
4. mySecondAlbum - trackThree
5. mySecondAlbum - trackFour
6. myFirstAlbum - trackTwo
7. myFirstAlbum - trackThree
8. myFirstAlbum - trackFour

### What is the expected _correct_ behavior?

The resulting playlist order on the mopidy device is: 

1. myFirstAlbum - trackOne (currently playing)
2. myFirstAlbum - trackTwo
3. myFirstAlbum - trackThree
4. myFirstAlbum - trackFour
5. mySecondAlbum - trackOne
6. mySecondAlbum - trackTwo
7. mySecondAlbum - trackThree
8. mySecondAlbum - trackFour

### Jellyfin Server version

10.9.8+

### Specify the build version

10.9.8

### Environment

```markdown
- OS: Debian 12.2.0-14
- Linux Kernel: 6.8.8-4-pve
- Virtualization: proxmox QEMU/KVM
- Clients: Browser
- Browser: Firefox 115.13.0-esr
- FFmpeg Version: 5.1.4-Jellyfin
- Playback Method: Direct Play to mopidy/gstreamer
- Hardware Acceleration: none
- GPU Model: none
- Plugins: default
- Reverse Proxy: none
- Base URL: none
- Networking: bridge
- Storage: NFS
```

### Jellyfin logs

```shell
[2024-08-02 10:43:35.261 -05:00] [WRN] PlaybackStopped reported with null media info.
[2024-08-02 10:49:57.204 -05:00] [WRN] PlaybackStopped reported with null media info.
[2024-08-02 10:50:04.780 -05:00] [INF] WS "192.168.55.10" closed
[2024-08-02 10:50:05.189 -05:00] [INF] WS "192.168.55.10" request
[2024-08-02 10:50:12.725 -05:00] [INF] GetPostedPlaybackInfo profile: DeviceProfile { Name: null, Id: null, MaxStreamingBitrate: 8000000, MaxStaticBitrate: 8000000, MusicStreamingTranscodingBitrate: 128000, MaxStaticMusicBitrate: 8000000, DirectPlayProfiles: [DirectPlayProfile { Container: "opus", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "mp3", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "aac", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "m4a", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "flac", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "webma", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "webm", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "wav", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "ogg", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "mpa", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "wma", AudioCodec: null, VideoCodec: null, Type: Audio }], TranscodingProfiles: [TranscodingProfile { Container: "mp3", Type: Audio, VideoCodec: "", AudioCodec: "aac", Protocol: hls, EstimateContentLength: False, EnableMpegtsM2TsMode: False, TranscodeSeekInfo: Auto, CopyTimestamps: False, Context: Streaming, EnableSubtitlesInManifest: False, MaxAudioChannels: null, MinSegments: 0, SegmentLength: 0, BreakOnNonKeyFrames: False, Conditions: [] }], ContainerProfiles: [], CodecProfiles: [], SubtitleProfiles: [] }
[2024-08-02 10:50:12.727 -05:00] [INF] User policy for "dacuser". EnableAudioPlaybackTranscoding: True
```

**Comment by AlbertMarkovski** [comment]
This bug persists in Jellyfin 10.10.6

**Comment by AlbertMarkovski** [comment]
Sorry if this was the wrong repo for this issue. 

Can this be transferred, or should I submit a new issue to Jellyfin-Web instead?

**Comment by theguymadmax** [comment]
This should be fixed by https://github.com/jellyfin/jellyfin-web/pull/6888 and will be included in the upcoming 10.11 release. If the issue still happens in 10.11, please open a new issue in the web repo.
